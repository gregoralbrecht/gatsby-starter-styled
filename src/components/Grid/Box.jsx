import styled, {
  space,
  color,
  layout,
  border,
  background,
  position,
  propTypes,
} from '@style'

const Box = styled.div`
  ${space}
  ${color}
  ${layout}
  ${border}
  ${background}
  ${position}
`

Box.displayName = `Box`

Box.propTypes = {
  ...propTypes.space,
  ...propTypes.color,
  ...propTypes.layout,
  ...propTypes.border,
  ...propTypes.background,
  ...propTypes.position,
}

export default Box
