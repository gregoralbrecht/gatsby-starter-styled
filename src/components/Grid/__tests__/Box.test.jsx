import React from 'react'
import { render } from '@test'
import Box from '../Box'
import 'jest-styled-components'

test('<Box> renders', () => {
  const { getByText } = render(<Box>Hello world</Box>)

  expect(getByText(`Hello world`)).toMatchSnapshot()
})

test('<Box> renders with props', () => {
  const { getByText } = render(
    <Box m={2} px={[1, 2, 3]}>
      Hello world
    </Box>
  )

  expect(getByText(`Hello world`)).toMatchSnapshot()
  expect(getByText(`Hello world`)).toHaveStyle(`margin: 8px`)
})

test('<Box> renders with as prop', () => {
  const { getByText } = render(
    <Box m={2} as="header">
      Hello world
    </Box>
  )

  expect(getByText(`Hello world`)).toMatchSnapshot()
  expect(getByText(`Hello world`)).toHaveStyle(`margin: 8px`)
})
