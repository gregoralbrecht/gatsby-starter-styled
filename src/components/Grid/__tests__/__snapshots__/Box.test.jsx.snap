// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box> renders 1`] = `
<div
  class=""
>
  Hello world
</div>
`;

exports[`<Box> renders with as prop 1`] = `
.c0 {
  margin: 8px;
}

<header
  class="c0"
>
  Hello world
</header>
`;

exports[`<Box> renders with props 1`] = `
.c0 {
  margin: 8px;
  padding-left: 4px;
  padding-right: 4px;
}

@media screen and (min-width:576px) {
  .c0 {
    padding-left: 8px;
    padding-right: 8px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

<div
  class="c0"
>
  Hello world
</div>
`;
